### PR TITLE
[BuildPipeline] Turn off publishing to myget

### PR DESCRIPTION
**Pull request by jashook**
For Coreclr, do not publish to myget, unless 'myget' is passed in PB_PublishType.

### CODE REVIEW

File: buildpipeline/DotNet-Trusted-Publish.json [hunk 1]
```diff
@@ -179,6 +179,7 @@
       "alwaysRun": false,
       "displayName": "packages -> dotnet.myget.org",
       "timeoutInMinutes": 0,
+      "condition": "eq(variables.PB_PublishType, 'myget'))",
```
Comment (karajas) [inline buildpipeline/DotNet-Trusted-Publish.json]: https://github.com/dotnet/corefx/blob/master/buildpipeline/DotNet-Trusted-Publish.json#L185 has a contains. May want something similar here

**Comment by jashook** [comment]
Updated, working on editing the build definitions now.

**Comment by jashook** [comment]
Updated.

File: buildpipeline/DotNet-Trusted-Publish.json [hunk 2]
```diff
@@ -179,6 +179,7 @@
       "alwaysRun": false,
       "displayName": "packages -> dotnet.myget.org",
       "timeoutInMinutes": 0,
+      "condition": "and(succeeded(), contains(variables.PB_PublishType, 'myget'), eq(ConfigurationGroup, 'Release'))",
```
Comment (weshaggard) [inline buildpipeline/DotNet-Trusted-Publish.json]: You need to at least also condition `Update versions repository`. Honestly I would go ahead and condition the blob publishing as well given it is trivial to add, but you can do that next week if you like.

**Comment by jashook** [comment]
Updated again ptal @weshaggard. Also thanks for the help and review

**Pull request merged**